core/lib/spree/core/state_machines/order.rb

```diff
@@ -0,0 +1,245 @@
+# frozen_string_literal: true
+
+module Spree
+  module Core
+    class StateMachines
+      module Order
```

Comment:
I don't think we can change the name of the module without a deprecation proxy. Anyway, I'm also ok leaving `Checkout` as its name for now. Is there a specific reason to change it within this PR?

Reply:
I'm resolving this conversation as I think it's not relevant anymore now that we decided to keep the old state machine around for a while.

@kennyadsl feel free to reopen if it makes sense to you.